src/main/java/com/google/sps/servlets/CaptionService.java

```diff
@@ -0,0 +1,18 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class CaptionService {
+  public InputStream fetchStream(String videoId, String language)
+      throws MalformedURLException, IOException {
+    if (videoId == null) {
+      throw new MalformedURLException();
```

Comment:
This should be `IllegalArgumentException` videoId is not a URL. Also should have exception message explaining what went wrong such as `videoId must be a non-null String value`.

Reply:
Done. Thanks for pointing it out!